website/content/docs/enterprise/namespaces/namespace-limits.mdx

```diff
@@ -0,0 +1,110 @@
+Vault **may be unresponsive** while this initialization is happening. Each
+namespace has at least the 4 default mounts (`sys`, `identity`, `cubbyhole`, and
+`token`) that need to be set up.
+
+Testing a varying number of namespaces, the following post-unseal times
+were observed:
+
+| Number of namespaces | Post-unseal time  |
+|----------------------|-------------------|
+| 10                   | ~5 seconds        |
+| 10000                | ~2-3 minutes      |
+| 20000                | ~12-14 minutes    |
+| 30000                | ~33-36 minutes    |
+
+This period of unresponsiveness can occur even when Vault is using auto-unseal.
+
+## Leadership transfers
+
+Leadership transfers take longer with more namespaces, and risk leaving the
+cluster in a leaderless state for longer.
+
+When testing a [`vault operator step-down`](/vault/docs/commands/operator/step-down)
+command, the following times were observed:
+
+| Number of namespaces | Time until a node is elected as leader |
+|----------------------|----------------------------------------|
+| 10                   | ~2 seconds                             |
+| 10000                | ~33-45 seconds                         |
+| 20000                | ~1-2 minutes                           |
+| 30000                | ~4 minutes                             |
+
+## Memory usage
+
+Each namespace requires at least 435 kilobytes of memory to store information
+about the paths available within the namespace. It is important to verify that
+the machines running Vault have enough memory to support the number of
+namespaces.
+
+## Rollback and periodic functions
+
+Once every minute, Vault will perform a rollback operation on every mount. By
+default, Vault will use 256 workers to perform these rollback operations. If
+some mounts take too long to complete the rollback operation, then the rollbacks
+can become backed up. The effects of this will vary based on the particular
+mounts that are being used. It may mean that stale data takes longer to be
+cleaned up, or that periodic rotations happen less frequently than intended.
+
+Backups in rollback operations can be detected by monitoring the
+[`vault.rollback.queued`](/vault/docs/internals/telemetry/metrics/core-system#rollback-metrics),
+and [`vault.rollback.waiting`](/vault/docs/internals/telemetry/metrics/core-system#rollback-metrics)
+metrics. If the number of queued items stays larger than 256, or if the waiting
+period is longer than 60 seconds, then the number of rollback workers isn't able
+to keep up.
+
+## Identity secret engine
+
+When using many namespaces, it is common to see warning level log lines from the
+root namespace's `identity` secret mount:
+
+```text
+2023-10-24T15:47:56.594Z [WARN]  secrets.identity.identity_51eb2411: error expiring OIDC public keys: err="context deadline exceeded"
+2023-10-24T15:47:56.594Z [WARN]  secrets.identity.identity_51eb2411: error rotating OIDC keys: err="context deadline exceeded"
+```
```

Comment:
````suggestion
When using OIDC with many namespaces, you may see warnings in your Vault logs
from the `identity` secret mount under the `root` namespace. For example:

```text
2023-10-24T15:47:56.594Z [WARN]  secrets.identity.identity_51eb2411: error expiring OIDC public keys: err="context deadline exceeded"
2023-10-24T15:47:56.594Z [WARN]  secrets.identity.identity_51eb2411: error rotating OIDC keys: err="context deadline exceeded"
```
````

Style correction: write in active voice